### PR TITLE
Stdlib: correct many parameter names

### PR DESCRIPTION
**Pull request by AlexWaygood**
I prepared this PR with the help of [this script](https://github.com/AlexWaygood/argcorrector) (which you'll notice has a very similar structure to `stubdefaulter`).

### CODE REVIEW

File: stdlib/multiprocessing/synchronize.pyi [hunk 1]
```diff
@@ -30,7 +30,7 @@ class Condition(AbstractContextManager[bool]):
     ) -> None: ...
 
 class Event:
-    def __init__(self, lock: _LockLike | None = ..., *, ctx: BaseContext) -> None: ...
+    def __init__(self, *, ctx: BaseContext) -> None: ...
```
Comment (AlexWaygood) [inline stdlib/multiprocessing/synchronize.pyi:33]: It doesn't look like this parameter has ever existed at runtime (from what I can see); I'm not sure how it made its way into the stub

File: stdlib/statistics.pyi [hunk 2]
```diff
@@ -123,9 +123,7 @@ if sys.version_info >= (3, 10):
         intercept: float
 
 if sys.version_info >= (3, 11):
-    def linear_regression(
-        __regressor: Sequence[_Number], __dependent_variable: Sequence[_Number], *, proportional: bool = False
-    ) -> LinearRegression: ...
+    def linear_regression(__x: Sequence[_Number], __y: Sequence[_Number], *, proportional: bool = False) -> LinearRegression: ...
```
Comment (hauntsaninja) [inline stdlib/statistics.pyi:126]: This one seems like a regression ;-)

File: stdlib/types.pyi [hunk 3]
```diff
@@ -589,7 +589,7 @@ if sys.version_info >= (3, 9):
         @property
         def __parameters__(self) -> tuple[Any, ...]: ...
         def __init__(self, origin: type, args: Any) -> None: ...
-        def __getitem__(self, __typeargs: Any) -> GenericAlias: ...
+        def __getitem__(self, __key: Any) -> GenericAlias: ...
```
Comment (hauntsaninja) [inline stdlib/types.pyi:592]: typeargs is more descriptive

**Review by hauntsaninja — COMMENTED** [review]
Looks good! Two cases where the old names are better than the new names

File: stdlib/statistics.pyi (hunk 2 above)
Comment (AlexWaygood) [reply stdlib/statistics.pyi:126]:
```suggestion
    def linear_regression(__regressor: Sequence[_Number], __dependent_variable: Sequence[_Number], *, proportional: bool = False) -> LinearRegression: ...
```

File: stdlib/statistics.pyi [hunk 4]
```diff
@@ -123,9 +123,7 @@ if sys.version_info >= (3, 10):
...
 
 elif sys.version_info >= (3, 10):
-    def linear_regression(__regressor: Sequence[_Number], __dependent_variable: Sequence[_Number]) -> LinearRegression: ...
+    def linear_regression(__x: Sequence[_Number], __y: Sequence[_Number]) -> LinearRegression: ...
```
Comment (AlexWaygood) [inline stdlib/statistics.pyi:129]:
```suggestion
    def linear_regression(__regressor: Sequence[_Number], __dependent_variable: Sequence[_Number]) -> LinearRegression: ...
```

File: stdlib/types.pyi (hunk 3 above)
Comment (AlexWaygood) [reply stdlib/types.pyi:592]:
```suggestion
        def __getitem__(self, __typeargs: Any) -> GenericAlias: ...
```

**Review by hauntsaninja — APPROVED** [review]

**Comment by AlexWaygood** [comment]
Thanks for the review!

**Pull request merged**